### PR TITLE
FIX: Many doc fixes

### PR DESCRIPTION
**Pull request by Eric89GXL**
A bunch of changes to make our docs and docstrings better. Also adds automated docstring checks.

### CODE REVIEW

File: doc/index.rst [hunk 1]
```diff
@@ -37,11 +41,12 @@ We are still working on a complete user guide for Vispy. In the meantime, you
 can:
 
   * Check out the `gallery <http://vispy.org/gallery.html>`_
-  * Use the ``mpl_plot`` experimental OpenGL backend for matplotlib
+  * Use the ``vispy.plot`` and ``vispy.scene`` interfaces for high-level work
```
Comment (campagnola) [inline doc/index.rst]: I thought single-backticks result in a link.. does that work for modules?

File: vispy/util/tests/test_docstring_parameters.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
+# TODO inspect for Cython (see sagenb.misc.sageinspect)
```
Comment (campagnola) [inline vispy/util/tests/test_docstring_parameters.py]: COOOOL.

**Comment by campagnola** [comment]
Can I describe this PR as "heroic" ? Yes, I think I just did.

File: doc/index.rst (hunk 1 above)
Comment (Eric89GXL) [inline doc/index.rst]: I tried and it didn't work for these unfortunately

**Comment by Eric89GXL** [comment]
Masochistic might be more appropriate :)

**Comment by campagnola** [comment]
Hey, Superman didn't really care either; he was just super anal. Still a hero.

**Comment by Eric89GXL** [comment]
Ready for review/merge @campagnola. Hopefully after this I can push a release tomorrow.

**Comment by Eric89GXL** [comment]
@campagnola feel free to take a last look and merge if you're happy -- I might be able to push the release out today

**Comment by campagnola** [comment]
Did we decide against implementing axes for this one?

**Comment by Eric89GXL** [comment]
Yeah. They won't work terribly well, and it will probably hold up release for to long. I think the idea is to push one out now, then make another one hopefully at the end of the summer with axes and collections

**Comment by campagnola** [comment]
Fine by me. 
Thanks!

**Comment by almarklein** [comment]
I know I am late, but I just have to say how much I like this :)

**Pull request merged**